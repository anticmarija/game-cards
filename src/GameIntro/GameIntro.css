.button {
  width: 33%;
  height: 40px;
  margin-bottom: 20px;
  border-radius: 20px;
  background-color: lightgreen;
  cursor: pointer;
  outline: none;
}

.button:hover {
  opacity: 0.8;
}
