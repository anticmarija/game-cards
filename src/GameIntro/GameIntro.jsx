import React from 'react';
import './GameIntro.css';
import { connect } from 'react-redux';
import { setNumberOfPlayers, fetchCards } from '../store/actions/game.actions';

export const GameIntro = ({ setNumberOfPlayers, fetchCards }) => {
  const POSSIBLE_NUMBERS = [2, 3, 4];
  return (
    <>
      <h1>Select number of players</h1>
      {POSSIBLE_NUMBERS.map(elem => (
        <button
          key={elem}
          className="button"
          onClick={() => {
            if (setNumberOfPlayers) {
              setNumberOfPlayers(elem);
            }
            if (fetchCards) {
              fetchCards(elem);
            }
          }}
        >
          {elem} players
        </button>
      ))}
    </>
  );
};

export const mapDispatchToProps = dispatch => ({
  setNumberOfPlayers: numberOfPlayers =>
    dispatch(setNumberOfPlayers(numberOfPlayers)),
  fetchCards: num => dispatch(fetchCards(num)),
});

export default connect(undefined, mapDispatchToProps)(GameIntro);
