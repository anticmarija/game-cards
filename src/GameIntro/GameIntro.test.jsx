import React from 'react';
import { render } from '@testing-library/react';
import { GameIntro, mapDispatchToProps } from './GameIntro';

describe('Test GameIntro component', () => {
  it('should render correctly', () => {
    const { container } = render(<GameIntro />);

    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('mapDispatchToProps', () => {
  const dispatchMock = jest.fn();

  it('should provide a prop to set mobile layout', () => {
    const dispatchProps = mapDispatchToProps(dispatchMock);

    dispatchProps.setNumberOfPlayers(2);
    expect(dispatchMock).toHaveBeenCalled();
  });
});
