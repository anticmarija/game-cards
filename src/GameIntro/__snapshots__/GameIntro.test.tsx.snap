// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GameIntro component should render correctly 1`] = `
<h1>
  Select number of players
</h1>
`;
